benchmarks/data_utils.py

```diff
@@ -0,0 +1,119 @@
+            path = os.path.join(path_to_folder, label)
+            files = os.listdir(path)
+            for file_name in files:
+                    if  name_constraint(file_name):
+                        self.vid_path.append(os.path.join(path, file_name))
+                        self.targets.append(target)
+        
+    def __len__(self):
+        return len(self.targets)
+
+    
+    def __getitem__(self, idx):
+        target_class = self.targets[idx]
+        vid = np.load(self.vid_path[idx])
+        first_frame = vid[0]
+        last_frame = vid[-1] 
+
+        first_frame = torch.as_tensor((first_frame))
+        last_frame = torch.as_tensor((last_frame))
+
+        # stack frames as channels, x shape is [2, img_width, img_height]
+        sample = torch.stack((first_frame, last_frame))
+
+       
+        if self.transform:
+            sample = self.transform(sample)
+            
+        return sample, target_class
+
+
+#TODO is this batchable, I think it is
+# TODO, return some indication that image is empty. 
+    #TODO - if it ends up empty, did protein move or dissapear?? how much do we handle at the moment
+#TODO also this is kinda slow if that becomes a problem.
+class RoiTransform:
+    """Pick a ROI based on pixel intensity. Choose the same coordinates for both/all frames"""
+
+    def __init__(self, window_size=(28,28), kernel_size=5, use_first_frame_values=True):
+        """
+        window_size = (width, height) of cropped section  
+        kernel_size = size of gaussian kernel used in selecting peak pixel values  
+        use_first_frame_values = if true, set ROI based off first frame, if false, set ROI based off last frame
+        """
+        self.width = window_size[0]
+        self.height = window_size[1]
+        self.kernel_size = kernel_size
+        self.use_first_frame = use_first_frame_values
+
+    def __call__(self, sample):
+        """
+        sample is [n_channels, img_width, img_height]
+        """
+
+        first_frame, last_frame = sample.split(1, dim=0)
+
+
+        if self.use_first_frame:
+            img_blurred = TF.gaussian_blur(first_frame, kernel_size=self.kernel_size) # TODO setting?
+        else:
+            img_blurred = TF.gaussian_blur(last_frame, kernel_size=self.kernel_size) # TODO setting?
+
+        # used the topk function in case we later decide to take multiple peaks
+        values, indices = torch.topk(img_blurred.flatten(), k=1)
+
+        """ 
+        Need to address the empty images (no mitochondria pictured) before train/test stage
+        """
+        # If first frame is empty, check the last frame for a signal
+        if values[0] == 0.0:
+            img_blurred = TF.gaussian_blur(last_frame, kernel_size=self.kernel_size) # TODO setting?
+            values, indices = torch.topk(img_blurred.flatten(), k=1)
+            if values[0] == 0.0:
+                print("NO DATA: maximum pixel value is {}, but this sample will still be trained on".format(values[0]))
```

Comment:
I prefer using `f-strings` since they're more readable and manageable in most cases, imo.
```suggestion
                print(f"NO DATA: maximum pixel value is {values[0]}, but this sample will still be trained on")
```

Reply:
what is this sorcery